Deprecate storage class overrides (#10407)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/ref-storage-persistent.adoc

== Storage class overrides

> WARNING: Storage class overrides are deprecated and will be removed in the future. Please use Kafka Node Pool resources instead.

Reviewer: ```suggestion
WARNING: Storage class overrides are deprecated and will be removed in the future. As a replacement, use `KafkaNodePool` resources instead.
```

The `overrides` property is currently used only to override the storage `class`. 
Overrides for other storage configuration properties is not currently supported.

> === Migrating from storage class overrides to Kafka Node Pools

Reviewer: ```suggestion
=== Migrating from storage class overrides to node pools
```

Storage class overrides are deprecated and will be removed in the future.

> If you use storage class overrides, you should use Kafka Node Pools instead.

Reviewer: ```suggestion
If you are using storage class overrides, we encourage you to transition to using node pools instead.
```

> To migrate the existing configuration, you should follow these steps:

Reviewer: ```suggestion
To migrate the existing configuration, follow these steps:
```
